crates/plex-api/src/media_container/server/library.rs

```diff
@@ -0,0 +1,496 @@
+use std::{fmt, str::FromStr};
+
+use monostate::MustBe;
+use serde::{
+    de::{self, Visitor},
+    Deserialize, Deserializer,
+};
+use serde_aux::prelude::{deserialize_number_from_string, deserialize_option_number_from_string};
+use serde_json::Value;
+use time::{Date, OffsetDateTime};
+
+use crate::media_container::MediaContainer;
+
+fn optional_boolish<'de, D>(deserializer: D) -> Result<Option<bool>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct BoolishVisitor;
+
+    impl<'de> Visitor<'de> for BoolishVisitor {
+        type Value = Option<bool>;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str("null or something that looks like a boolean")
+        }
+
+        fn visit_none<E>(self) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            Ok(None)
+        }
+
+        fn visit_i64<E>(self, v: i64) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            if v == 0 {
+                Ok(Some(false))
+            } else {
+                Ok(Some(true))
+            }
+        }
+
+        fn visit_u64<E>(self, v: u64) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            if v == 0 {
+                Ok(Some(false))
+            } else {
+                Ok(Some(true))
+            }
+        }
+
+        fn visit_str<E>(self, val: &str) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            if val == "1" {
+                Ok(Some(true))
+            } else {
+                Ok(Some(false))
+            }
+        }
+
+        fn visit_bool<E>(self, v: bool) -> Result<Self::Value, E>
+        where
+            E: de::Error,
+        {
+            Ok(Some(v))
+        }
+    }
+
+    let visitor = BoolishVisitor;
+    deserializer.deserialize_any(visitor)
+}
+
+#[derive(Debug, Clone, Deserialize, enum_utils::FromStr)]
```

Comment:
I've been using `derive_fromstr_from_deserialize!()` for generating FromStr based on the deserializer :D I think it'll be a good idea to replace usages of that `derive_fromstr_from_deserialize` to `enum_utils::FromStr` later.